azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -470,45 +472,81 @@ def disable(self, reason, disable_cgroups):
             except Exception as exception:
                 log_cgroup_warning("Error disabling cgroups: {0}".format(ustr(exception)))
 
-        def _set_cpu_quota(self, unit_name, quota):
-            """
-            Sets CPU quota to the given percentage (100% == 1 CPU)
+        def _get_resource_quotas(self):
+            if self._cgroups_api is None:
+                return []
+            return [CpuQuota(self._cgroups_api), MemoryQuota(self._cgroups_api)]
 
-            NOTE: This is done using a systemtcl set-property --runtime; any local overrides in /etc folder on the VM will take precedence
-            over this setting.
+        def _set_resource_quota(self, unit_name, resource_type, quota):
```

Comment:
Would it make more sense for quota to always be a dict? If only setting one thing between cpu or memory, then there would only be one value in the dict. That way we don't need to pass ResourceName in either

Reply:
not sure If I followed. If it always dict and no resouceName as the key, then how that works?